transport/transport.go

```diff
@@ -306,3 +319,71 @@ func NewMultiTransport(backends []http.RoundTripper,
 		Backends:        backends,
 		HandleResponses: responsesHandler}
 }
+
+// SetTransportsConfig assign transport config to Matcher
+func (m *Matcher) SetTransportsConfig(clientConfig httphandlerConfig.Client) {
+	m.TransportsConfig = clientConfig.Transports
+}
+
+// SelectTransport returns transport instance by method, path and queryParams
+func (m *Matcher) SelectTransport(method, path, queryParams string) (matchedTransport config.Transport) {
+	matchedTransport, ok := m.TransportsConfig.GetMatchedTransport(method, path, queryParams)
+	if !ok {
+		log.DefaultLogger.Fatalf("Transport not matched with args. method: %s, path: %s, queryParams: %s", method, path, queryParams)
+	}
+	return
+}
+
+// RoundTrip for transport matching
+func (m *Matcher) RoundTrip(request *http.Request) (*http.Response, error) {
+	return m.SelectTransportRoundTripper(request).RoundTrip(request)
+}
+
+// SelectTransportRoundTripper for selecting RoundTripper by request object from transports matcher
+func (m *Matcher) SelectTransportRoundTripper(request *http.Request) (selectedRoundTripper http.RoundTripper) {
+	selectedTransport := m.SelectTransport(request.Method, request.URL.Path, request.URL.RawQuery)
+	reqID := request.Context().Value(log.ContextreqIDKey)
+	log.DefaultLogger.Debugf("Request %s - selected transport name: %s (by method: %s, path: %s, queryParams: %s)",
+		reqID, selectedTransport.Name, request.Method, request.URL.Path, request.URL.RawQuery)
+
+	return m.RoundTrippers[selectedTransport.Name]
```

Comment:
Does this element differ from `selectedTransport`?

Reply:
If not, we should consider removing Matcher.RoundTrippers map

Reply:
Ok. I see where the problem is... it's the name of variable. For readability sakes: rename config.Transport to something that would not be confused with http.Transport and rename methods accordingly

Reply:
@mjarco Proposals: `TransportProperties`, `TransportClientProps`, `HttpTransportProps` ... which name is the best for you ?

Reply:
`Kind`

Reply:
`TransportRules`?

Reply:
apply rules are properties of this type